wordpress-mobile/WordPress-Android
# Fixing NoSuchElement list empty in PostDayViewsUseCase

Fixes #10830

This PR does two things:

1. Attempt to fix issue #10830: was not actually able to reproduce but applied same approach as used in `OverviewUseCase.buildUiModel` (that is introduced a check for the list not being empty)
2. In `PostDayViewsUseCase.fetchRemoteData` in case of error I used `selectedDateProvider.onDateLoadingFailed` insetad of `selectedDateProvider.onDateLoadingSucceeded`

## To test:

Since it was not possible to reproduce, I think we can only make sure we did not introduce issues in the relevant stats card, so smoke test the post day/views card with different sites/posts and compare with the not modified WordPress app to check you get the same visualizations.

PR submission checklist:
